test/extended/builds/metrics.go

```diff
@@ -0,0 +1,323 @@
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	o.Expect(err).NotTo(o.HaveOccurred())
+	debug := ""
+	debug = debug + fmt.Sprintf("\n\nstatus %d body: \n\n%s\n\n", resp.StatusCode, string(body))
+	if resp.StatusCode != 200 {
+		fmt.Fprintf(g.GinkgoWriter, debug)
+	}
+	o.Expect(resp.StatusCode).To(o.Equal(200))
+
+	result := PrometheusResponse{}
+	json.Unmarshal(body, &result)
+	debug = debug + fmt.Sprintf("\nunmarshall json status %s\n data: %s\n", result.Status, result.Data)
+	metrics := result.Data.Result
+
+	foundCorrectMetricLabelsAndValue := false
+	for _, sample := range metrics {
+		debug = debug + fmt.Sprintf("\nmetric %s with value %s\n", sample.Metric.String(), sample.Value.String())
+
+		// first see if a metric has all the label names and label values we are looking for
+		foundCorrectLabels := true
+		for labelName, labelValue := range tc.labels {
+			debug = debug + fmt.Sprintf("\nlooking for label %s", labelName)
+			if v, ok := sample.Metric[model.LabelName(labelName)]; ok {
+				debug = debug + fmt.Sprintf("\nfound label %s  set to %s when looking for %s\n", labelName, v, labelValue)
+				if string(v) != labelValue {
+					foundCorrectLabels = false
+					break
+				}
+			} else {
+				foundCorrectLabels = false
+				break
+			}
+		}
+
+		// if found metric with correct set of labels, now see if the metric value is what we are expecting
+		if foundCorrectLabels {
+			switch {
+			case len(tc.equals) > 0:
+				debug = debug + fmt.Sprintf("\ntesting %s == %s\n", sample.Value, tc.equals)
+				if x, err := strconv.ParseFloat(tc.equals, 64); err == nil && float64(sample.Value) == x {
+					foundCorrectMetricLabelsAndValue = true
+				}
+			case len(tc.greaterThan) > 0:
+				debug = debug + fmt.Sprintf("\ntesting %s > %s\n", sample.Value, tc.greaterThan)
+				if x, err := strconv.ParseFloat(tc.greaterThan, 64); err == nil && float64(sample.Value) > x {
+					foundCorrectMetricLabelsAndValue = true
+				}
+			}
+			break
+		}
+
+	}
+	if !foundCorrectMetricLabelsAndValue {
+		fmt.Fprintf(g.GinkgoWriter, debug)
+	}
+	o.Expect(foundCorrectMetricLabelsAndValue).To(o.BeTrue())
+}
+
+var _ = g.Describe("[builds][Slow] openshift build prometheus metrics", func() {
+	defer g.GinkgoRecover()
+	var (
+		oc                     = exutil.NewCLI("bld-mtrcs", exutil.KubeConfigPath())
+		appPrometheusReady     = false
+		appPrometheusNamespace = ""
+		counterTest            = MetricTest{
+			metric: bldmet.TerminalBuildCountQuery,
+			labels: map[string]string{"phase": "complete"},
+			equals: "1",
+		}
+		newGaugeTest = MetricTest{
+			metric: bldmet.ActiveBuildCountQuery,
+			labels: map[string]string{"phase": "new"},
+			equals: "0",
+		}
+		pendingGaugeTest = MetricTest{
+			metric: bldmet.ActiveBuildCountQuery,
+			labels: map[string]string{"phase": "pending"},
+			equals: "0",
+		}
+		runningGaugeTest = MetricTest{
```

Comment:
I liked the grouping / readability of it.  But yes, a more limited scope for them would work.  I'll move them.